src/wp-includes/revision.php

```diff
@@ -361,15 +387,36 @@ function _wp_put_post_revision( $post = null, $autosave = false ) {
 		 * Fires once a revision has been saved.
 		 *
 		 * @since 2.6.0
+		 * @since 6.4.0 The post_id parameter was added.
 		 *
 		 * @param int $revision_id Post revision ID.
+		 * @param int $post_id     Post ID.
 		 */
-		do_action( '_wp_put_post_revision', $revision_id );
+		do_action( '_wp_put_post_revision', $revision_id, $post['post_parent'] );
 	}
 
 	return $revision_id;
 }
 
+
+/**
+ * Save the revisioned meta fields.
+ *
+ * @param int $revision_id The ID of the revision to save the meta to.
+ * @param int $post_id     The ID of the post the revision is associated with.
+ *
+ * @since 6.4.0
+ */
+function wp_save_revisioned_meta_fields( $revision_id, $post_id ) {
+
+	// Save revisioned meta fields.
+	foreach ( wp_post_revision_meta_keys( get_post_type( $post_id ) ) as $meta_key ) {
```

Comment:
get_post_type( $post_id ) could return false. Error handling?

Reply:
I'm not sure about this one. Under what conditions would the post_type be false? Would we want to skip the check entirely?

I assume we would exit early on an error here, however do we need to? Passing false to the `wp_post_revision_meta_keys` function will still work, it will use meta types from 'post' and pass the `false` type to the filter.

Reply:
It would be `false` if `get_post()` returns a falsey value. That would typically happen the post ID is falsey or that ID doesn't exist.

In the normal run, the new function is a hooked callback and the post ID is from the post's array. That said, is there a possible scenario where the `'post_parent'` post does not exist in the database?

Also what happens if this function is directly called by an extender? Safeguarding would protect against a falsey state.

Reply:
```
foreach ( wp_post_revision_meta_keys( get_post_type( $post_id ) )
```

I'd also suggest breaking out the compound code from the `foreach()` to make the code more understandable for all contributors.

```php
$post_type = get_post_type( $post_id );
if ( ! $post_type ) {
   return;
}

foreach ( wp_post_revision_meta_keys( $post_type as $meta_key )  {
```

Reply:
Makes sense, thanks for the feedback.

Reply:
Note, also updating in `wp_restore_post_revision_meta()`

Reply:
updated